scripts/src/image-browse.js

```diff
@@ -4,13 +4,15 @@ import calc_scroll_percent_increment from "./modules/analytics/scroll_tracking/c
 import update_scroll_obj from "./modules/analytics/scroll_tracking/update_scroll_obj";
 import get_scroll_percentage from "./modules/analytics/scroll_tracking/get_scroll_percentage";
 
-add_data_link('.image-browse__listing a');
-
 let scrollObj = {
     'event': 'image-viewer-browse',
     highestScrollPercentage: get_scroll_percentage()
 };
 
+document.addEventListener('DOMContentLoaded', () => {
+    add_data_link('.image-browse__listing a');
+});
+
 document.addEventListener('scroll', e => {
```

Comment:
Just out of curiosity, would we want to move everything inside `DOMContentLoaded`? I think `calc_scroll_percent_increment` uses `document` to calculate the height and I think it's also called in `get_scroll_percentage` under certain conditions. I'm probably wrong but was just wondering.

Reply:
I think it's OK as it is. Think of `DOMContentLoaded` as an event, whereas `calc_scroll_percent_increment` and `get_scroll_percentage` will access `document`, they'll do in different contexts. We also need access to `scrollObj` from within the `scroll` event handler. Happy to discuss this though if you'd like to.